src/runtime/create-runtime-stories.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-env browser */
+import { logger } from '@storybook/client-logger';
+import type { Meta, StoryObj } from '@storybook/svelte';
+import { mount, unmount, type ComponentType } from 'svelte';
+
+import StoriesExtractor from './StoriesExtractor.svelte';
+import StoryRenderer from './StoryRenderer.svelte';
+import type { StoriesRepository } from './contexts/extractor.svelte.js';
+
+const createFragment = document.createDocumentFragment
+  ? () => document.createDocumentFragment()
+  : () => document.createElement('div');
+
+/**
+ * @module
+ * Called from a bundler.
+ *
+ * It mounts the Stories components in a context which disables
+ * the rendering of every `<Story />`,
+ * but instead collects names and properties.
+ *
+ * For every discovered `<Story />`, it creates a `StoryFn` which
+ * instantiate the main Stories component: Every Story but
+ * the one selected is disabled.
+ */
+// TODO: I'm not sure the 'meta' is necessary here. As long as it's default exported, SB should internally combine it with the stories. Except for the play logic below, that looks funky, need to ask Pablo about that.
+export const createRuntimeStories = <TMeta extends Meta>(Stories: ComponentType, meta: TMeta, codeByStoryMap: Record<string, string>) => {
+  const repository: StoriesRepository<TMeta> = {
+    stories: new Map(),
+  };
+
+  try {
+    const context = mount(StoriesExtractor, {
+      target: createFragment() as Element,
+      props: {
+        Stories,
+        repository: () => repository,
+      },
+    });
+
+    unmount(context);
+  } catch (e: any) {
+    logger.error(`Error in mounting stories ${e.toString()}`, e);
+  }
+
+  const stories: Record<string, StoryObj<StoryRenderer<TMeta>>> = {};
+
+  for (const [exportName, story] of repository.stories) {
+    const storyObj: StoryObj<StoryRenderer<TMeta>> = {
+      ...story,
+      render: (args, storyContext) => ({
+        Component: StoryRenderer<TMeta>,
+        props: {
+          exportName,
+          Stories,
+          code: codeByStoryMap[exportName],
```

Comment:
We don't want to use `story.parameters.docs.source.code` at all if we can help it, because that is supposed to be a user-facing API. Setting that means that the user can't use `parameters.docs.source.transform` to transform our original string, because `code` cancels out `transform`.

https://storybook.js.org/docs/api/doc-block-source#code

https://storybook.js.org/docs/api/doc-block-source#transform

Reply:
perhaps we can set it at `story.parameters.__svelteCsf.rawCode` instead? I'm fine with that solution, and then removing the whole `codeByStoryMap` logic that I wrote.

I've already seen the need for `story.parameters.__svelteCsf = true` because I want `@storybook/svelte` to handle Svelte CSF stories differently, so having that parameter is good I think.

Reply:
Hmm. Knowing now, that we don't want to use `story.parameters.docs.source.code`, do you still think it's the right architecture to modify the story parameters instead of adding it at the bottom of the file the way I've "proposed" it?

Which is the simpler implementation? Any pros or cons?

I'm not saying one way is better than the other, but I'd like to hear your thoughts.

Reply:
In the current moment, I can say that adding a new property to `parameters` is the easiest way to resolve my misdirection.

**Pros:**

1. Easier to organize/group together along with other necessary properties strictly for this addon only _(`__svelteCsf`)_.
2. Following above, easier to access everything related to this addon at runtime, and control it.
3. (Maybe?) Easier to understand the AST wizardry we're going in the post-transform. Less AST nodes needed to create manually, since we're going to put everything in one `ObjectExpression` - `parameters.__svelteCsf`

**Cons**:

I can't think of any for the moment.

Reply:
Great! Sounds good to me